### PR TITLE
[flink] support longtext data type mapping bytes

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
### Purpose

support longtext data type mapping bytes

### CODE REVIEW

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTypeUtils.java [hunk 1]
```diff
@@ -262,6 +263,9 @@ public static DataType toDataType(
             case TEXT:
             case MEDIUMTEXT:
             case LONGTEXT:
+                return length == null || length == 0 || typeMapping.containsMode(LONGTEXT_TO_BYTES)
+                        ? DataTypes.VARCHAR(length)
+                        : DataTypes.STRING();
```
Comment (zhuangchong) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTypeUtils.java:268]: DataTypes.VARBINARY?

**Review by zhuangchong — CHANGES_REQUESTED** [review]

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTypeUtils.java (hunk 1 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTypeUtils.java:268]: Sorry, this is WIP and the code has not been submitted.

Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTypeUtils.java:268]: There's an issue with my VPN, so I won't be able to submit the code today.😭

**Comment by MonsterChenzhuo** [comment]
@yuzelin @zhuangchong  PTAL,thanks.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/TypeMapping.java [hunk 2]
```diff
@@ -70,6 +70,7 @@ public enum TypeMappingMode {
         TINYINT1_NOT_BOOL,
         TO_NULLABLE,
         TO_STRING,
+        LONGTEXT_TO_BYTES,
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/TypeMapping.java:73]: Update comment above `TypeMappingMode`.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTypeUtils.java [hunk 3]
```diff
@@ -286,6 +286,10 @@ public static DataType toDataType(
                 return DataTypes.STRING();
                 // MySQL BINARY and VARBINARY are stored as bytes in JSON. We convert them to
                 // DataTypes.VARBINARY to retain the length information
```
[the remark this answers is not in the document]
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTypeUtils.java]: done.

**Review by yuzelin — APPROVED** [review]
+1

**Pull request merged**